lib/client.js

```diff
@@ -22,7 +22,11 @@ var clientOptions = require('./client-options');
  *  Only one contact point is required (the driver will retrieve the address of the other nodes automatically),
  *  but it is usually a good idea to provide more than one contact point, because if that single contact point is unavailable, the driver will not be able to initialize correctly.
  * </p>
- * @property {String} keyspace
+ * @property {String} keyspace The logged keyspace for all the connections created within the {@link Client} instance.
+ * @property {Number} refreshSchemaDelay The default window size in milliseconds used to debounce node list and schema
+ * refresh metadata requests. Default: 1000.
+ * @property {Boolean} isMetadataEnabled Determines whether client-side schema metadata retrieval and update is enabled.
```

Comment:
As you point out, it looks like the `isMetadataEnabled` flag implementation is incorrect.

When automatic metadata retrieval is not enabled the driver: 

1. Should NOT retrieve keyspaces metadata.
1. Should NOT cache results of calls to get metadata of a schema object,  ie: `Metadata#getTable()` should always get the latest info.
1. Should get topology information (on init and up to date based on topology events).
1. Should build tokens: [tokens are retrieved and primary replicas][1] are populated but keyspace-specific replicas per token are [lazily calculated when the first call to `getReplicas()` for a given keyspace is made][2]. This could still be valid with metadata retrieval is disabled.

`Metadata#refreshKeyspaces()` is a public method so users can call it manually if they want to get/refresh the keyspaces information.

Let me know if I you agree with the above reasoning. In the meantime, I'll push a new commit to try it out.

[1]: https://github.com/datastax/nodejs-driver/blob/74a48f7bf7f786041615eea6bb70e1ae494c5b9a/lib/metadata/index.js#L90
[2]: https://github.com/datastax/nodejs-driver/blob/74a48f7bf7f786041615eea6bb70e1ae494c5b9a/lib/metadata/index.js#L202